### PR TITLE
PERF: Add var to masked arrays

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/array_algos/masked_reductions.py [hunk 1]
```diff
@@ -149,3 +150,19 @@ def mean(
...
+    *,
+    skipna: bool = True,
+    axis: int | None = None,
+    ddof: int = 1,
+):
+    if not values.size or mask.all():
+        return libmissing.NA
+
+    return _reductions(
+        np.var, values=values, mask=mask, skipna=skipna, axis=axis, **{"ddof": ddof}
```
Comment (mroeschke) [inline pandas/core/array_algos/masked_reductions.py:167]:
```suggestion
        np.var, values=values, mask=mask, skipna=skipna, axis=axis, ddof=ddof
```
?

File: pandas/tests/reductions/test_reductions.py [hunk 2]
```diff
@@ -787,6 +787,15 @@ def test_mean_masked_overflow(self):
         assert result_masked - result_numpy == 0
         assert result_masked == 1e17
 
+    def test_var_masked_array(self):
```
Comment (mroeschke) [inline pandas/tests/reductions/test_reductions.py:790]: Could you parameterize over `ddof=[0, 1]`?

File: pandas/core/array_algos/masked_reductions.py (hunk 1 above)
Comment (phofl) [reply pandas/core/array_algos/masked_reductions.py:167]: Ah thx, I tend to mess this up...

File: pandas/tests/reductions/test_reductions.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/reductions/test_reductions.py]: added

**Review by mroeschke — APPROVED** [review]
LGTM. Can merge on green after the merge conflict is resolved

**Comment by rhshadrach** [comment]
This patch may have induced a potential regression. Please check the links below. If any ASVs are parameterized, the combinations of parameters that a regression has been detected appear as subbullets. This is a partially automated message.

 - https://asv-runner.github.io/asv-collection/pandas/#series_methods.NanOps.time_func
   - func='var'; N=1000000; dtype='boolean'

**Comment by rhshadrach** [comment]
Here is a more direct link (still need to workout how to incorporate parameterizations in the link):

https://asv-runner.github.io/asv-collection/pandas/#series_methods.NanOps.time_func?p-func='var'&p-dtype='boolean'&p-N=1000000

**Comment by phofl** [comment]
Hm guess boolean is special casy here. We could cast the array to float, this would get most of the regression back, but not sure if this is worth it

**Comment by rhshadrach** [comment]
Taking the variance of Booleans seems pretty rare to me, but at the same time, the solution sounds very easy. I'm thinking to at least open a tracking issue.

**Pull request merged**